File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/context/roleplay/party/PartyGuestInformations.scala





















// Generated on 05/03/2013 10:35:05
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait PartyGuestInformations extends TypeDeserializer {
  val typeId = 374

  protected trait Type extends BaseType with Serializable {
    val typeId = PartyGuestInformations.typeId

    val guestId: Int
    val hostId: Int
    val name: String
    val guestLook: EntityLook.TypeImpl
    val breed: Byte
    val sex: Boolean

    def serialize(buf: Buffer) {
      buf.writeInt(guestId)
      buf.writeInt(hostId)
      buf.writeString(name)
      guestLook.serialize(buf)
      buf.writeByte(breed)
      buf.writeBoolean(sex)
    }

  }

  protected def create(guestId: Int, hostId: Int, name: String, guestLook: EntityLook.TypeImpl, breed: Byte, sex: Boolean): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val guestId = buf.readInt()
    val hostId = buf.readInt()
    val name = buf.readString()
    val guestLook = EntityLook.deserialize(buf)
    val breed = buf.readByte()
    val sex = buf.readBoolean()

    create(guestId, hostId, name, guestLook, breed, sex)
  }

}

object PartyGuestInformations extends PartyGuestInformations {
  case class TypeImpl(guestId: Int, hostId: Int, name: String, guestLook: EntityLook.TypeImpl, breed: Byte, sex: Boolean) extends Type
  override def create(guestId: Int, hostId: Int, name: String, guestLook: EntityLook.TypeImpl, breed: Byte, sex: Boolean) = TypeImpl(guestId, hostId, name, guestLook, breed, sex)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(guestId: Int, hostId: Int, name: String, guestLook: EntityLook.TypeImpl, breed: Byte, sex: Boolean) = create(guestId, hostId, name, guestLook, breed, sex)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/PartyGuestInformations.java


// Generated on 05/08/2013 19:38:01
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class PartyGuestInformations extends NetworkType {
    public static final short TYPE_ID = 374;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public int guestId;
    public int hostId;
    public String name;
    public EntityLook guestLook;
    public byte breed;
    public boolean sex;
    public PlayerStatus status;
    
    public PartyGuestInformations() { }
    
    public PartyGuestInformations(int guestId, int hostId, String name, EntityLook guestLook, byte breed, boolean sex, PlayerStatus status) {
        this.guestId = guestId;
        this.hostId = hostId;
        this.name = name;
        this.guestLook = guestLook;
        this.breed = breed;
        this.sex = sex;
        this.status = status;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(guestId);
        buf.writeInt(hostId);
        buf.writeString(name);
        guestLook.serialize(buf);
        buf.writeByte(breed);
        buf.writeBoolean(sex);
        buf.writeShort(status.getTypeId());
        status.serialize(buf);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        guestId = buf.readInt();
        if (guestId < 0)
            throw new RuntimeException("Forbidden value on guestId = " + guestId + ", it doesn't respect the following condition : guestId < 0");
        hostId = buf.readInt();
        if (hostId < 0)
            throw new RuntimeException("Forbidden value on hostId = " + hostId + ", it doesn't respect the following condition : hostId < 0");
        name = buf.readString();
        guestLook = new EntityLook();
        guestLook.deserialize(buf);
        breed = buf.readByte();
        sex = buf.readBoolean();
        status = ProtocolTypeManager.getInstance().build(buf.readShort());
        status.deserialize(buf);
    }
    
}
